src/Microsoft.Data/DataFrame.cs

```diff
@@ -133,13 +141,125 @@ public DataFrame Sort(string columnName, bool ascending = true)
+                for (long i = 0; i < minLength; i++)
+                {
+                    mapIndices[i] = i;
+                }
+                for (int i = 0; i < ColumnCount; i++)
+                {
+                    BaseColumn newColumn;
+                    if (RowCount < other.RowCount)
+                    {
+                        newColumn = Column(i).Clone(numberOfNullsToAppend: other.RowCount - RowCount);
+                    }
+                    else
+                    {
+                        newColumn = Column(i).Clone(mapIndices);
+                    }
+                    ret.InsertColumn(ret.ColumnCount, newColumn);
+                }
+                for (int i = 0; i < other.ColumnCount; i++)
+                {
+                    BaseColumn newColumn = other.Column(i).Clone();
+                    SetSuffixForDuplicatedColumnNames(ret, newColumn, leftSuffix, rightSuffix);
+                    ret.InsertColumn(ret.ColumnCount, newColumn);
+                }
+            }
+            else if (joinAlgorithm == JoinAlgorithm.Outer)
+            {
+                long newRowCount = Math.Max(RowCount, other.RowCount);
+                long numberOfNulls = newRowCount - RowCount;
+                for (int i = 0; i < ColumnCount; i++)
+                {
+                    BaseColumn newColumn = Column(i).Clone(numberOfNullsToAppend: numberOfNulls);
+                    ret.InsertColumn(ret.ColumnCount, newColumn);
+                }
+                numberOfNulls = newRowCount - other.RowCount;
+                for (int i = 0; i < other.ColumnCount; i++)
+                {
+                    BaseColumn newColumn = other.Column(i).Clone(numberOfNullsToAppend: numberOfNulls);
+                    SetSuffixForDuplicatedColumnNames(ret, newColumn, leftSuffix, rightSuffix);
+                    ret.InsertColumn(ret.ColumnCount, newColumn);
+                }
+            }
+            else if (joinAlgorithm == JoinAlgorithm.Inner)
+            {
+                long newRowCount = Math.Min(RowCount, other.RowCount);
+                PrimitiveColumn<long> mapIndices = new PrimitiveColumn<long>("mapIndices", newRowCount);
+                for (long i = 0; i < newRowCount; i++)
+                {
+                    mapIndices[i] = i;
+                }
+                for (int i = 0; i < ColumnCount; i++)
+                {
+                    BaseColumn newColumn = Column(i).Clone(mapIndices);
+                    ret.InsertColumn(ret.ColumnCount, newColumn);
+                }
+                for (int i = 0; i < other.ColumnCount; i++)
+                {
+                    BaseColumn newColumn = other.Column(i).Clone(mapIndices);
+                    SetSuffixForDuplicatedColumnNames(ret, newColumn, leftSuffix, rightSuffix);
+                    ret.InsertColumn(ret.ColumnCount, newColumn);
+                }
+            }
+            return ret;
```

Comment:
If someone specifies an invalid `joinAlgorithm` value, we just return an empty DataFrame. Is that intended behavior?

Reply:
Fixed